tests/legacy-cli/e2e/tests/basic/update-7.0.ts

```diff
@@ -0,0 +1,46 @@
+import { createProjectFromAsset } from '../../utils/assets';
+import { ng, silentNpm } from '../../utils/process';
+import {
+  isPrereleaseCli, removeHttpDep, useBuiltPackages, useCIChrome, useCIDefaults,
+} from '../../utils/project';
+import { expectFileToMatch } from '../../utils/fs';
+
+
+export default async function () {
+  const extraUpdateArgs = await isPrereleaseCli() ? ['--next', '--force'] : [];
+
+  // Create new project from previous version files.
+  // We must use the original NPM packages to force a real update.
+  // This also means that we can only truly test updates after a release.
+  await createProjectFromAsset('7.0-project', true);
+  // Update the CLI.
+  await ng('update', '@angular/cli', ...extraUpdateArgs);
+  // Should update the lazy route syntax.
+  await expectFileToMatch('src/app/app-routing.module.ts',
+  `loadChildren: () => import('./lazy/lazy.module').then(m => m.LazyModule)`);
+  // Should update the build-angular version.
+  await expectFileToMatch('src/app/app-routing.module.ts',
+    `"@angular-devkit/build-angular": "~0.800"`);
+  // Use the packages we are building in this commit, and CI Chrome.
+  await useBuiltPackages();
+  await useCIChrome('src/');
+  await useCIChrome('e2e/');
+  await useCIDefaults('seven-oh-project');
+  await silentNpm('install');
+  // TODO: Remove http? Should this be part of the migration? Or even fail it.
+  await removeHttpDep();
```

Comment:
Blocked on figuring out what to do about the http dep. It was added automatically in all releases up to now, so most projects have it.

Reply:
We should just always remove `@angular/http` from users package.json.
We can add more messaging around that for RC

Reply:
PR at https://github.com/angular/angular-cli/pull/14250